pkg/clusteragent/externalmetrics/model/datadogmetricinternal.go

```diff
@@ -106,9 +109,20 @@ func NewDatadogMetricInternalFromExternalMetric(id, query, metricName, autoscale
 	}
 }
 
-// UpdateFrom updates the `DatadogMetricInternal` from `DatadogMetric` Spec, returns modified instance
+// UpdateFrom updates the `DatadogMetricInternal` from `DatadogMetric` Spec
 func (d *DatadogMetricInternal) UpdateFrom(currentSpec datadoghq.DatadogMetricSpec) {
+	if d.shouldResolveQuery(currentSpec) {
```

Comment:
With comment below, I think it could be simplified to:

```
if shouldResolve() {
  resolveQuery()
}

d.query = currentSpec.query
```

Moving `Query` to internal field too and using the `Query()` accessor.